anomaly/config.cpp

```diff
@@ -0,0 +1,53 @@
+#include "config.hpp"
+
+#include <thirdparty/jsonxx/jsonxx.h>
+#include <thirdparty/jsonxx/jsonxx_utils.h>
+#include <fstream>
+
+namespace HawkTracer
+{
+namespace anomaly
+{
+
+bool Config::load_from_file(const std::string& file_name)
+{
+    std::ifstream file;
+    file.open(file_name);
+    if (!file.is_open())
+    {
+        return false;
+    }
+    _load_file(file);
+    return true;
+}
+
+unsigned int Config::get_insert_cost()
+{
+    return _insert_cost;
+}
+
+unsigned int Config::get_delete_cost()
+{
+    return _delete_cost;
+}
+
+unsigned int Config::get_relabel_cost(std::string& src, std::string& dst)
```

Comment:
it should be `const std::string&` so you can also pass r-values to the function.